dataRetriever/resolvers/peerAuthenticationResolver.go

```diff
@@ -142,59 +151,171 @@ func (res *peerAuthenticationResolver) ProcessReceivedMessage(message p2p.Messag
 	return err
 }
 
-func (res *peerAuthenticationResolver) resolveOneHash(hash []byte, chunkIndex int, pid core.PeerID) error {
-	peerAuthMsgs := res.fetchPeerAuthenticationMessagesForHash(hash)
-	if len(peerAuthMsgs) == 0 {
+// resolveChunkRequest sends the response for a chunk request
+func (res *peerAuthenticationResolver) resolveChunkRequest(chunkIndex int, epoch uint32, pid core.PeerID) error {
+	sortedPKs, err := res.getSortedValidatorsKeys(epoch)
+	if err != nil {
+		return err
+	}
+	if len(sortedPKs) == 0 {
 		return nil
 	}
 
-	if len(peerAuthMsgs) > maxNumOfPeerAuthenticationInResponse {
-		return res.sendMessageFromChunk(hash, peerAuthMsgs, chunkIndex, pid)
+	maxChunks := res.getMaxChunks(sortedPKs)
+	pksChunk, err := res.extractChunk(sortedPKs, chunkIndex, res.maxNumOfPeerAuthenticationInResponse, maxChunks)
+	if err != nil {
+		return err
+	}
+
+	var lastErr error
+	errorsFound := 0
+	dataSlice := make([][]byte, 0, res.maxNumOfPeerAuthenticationInResponse)
+	for _, pk := range pksChunk {
+		peerAuth, tmpErr := res.fetchPeerAuthenticationAsByteSlice(pk)
+		if tmpErr != nil {
```

Comment:
I do not think we should add the errors here. It will just clutter the node in normal circumstances when some peer authentication messages are missing. Instead, I would return an error if the data slice is empty, outside this for loop